test/lib/test_utils.ml

```diff
@@ -0,0 +1,115 @@
+open Ppxlib
+open OUnit2
+
+module Fixtures = struct
+  let location =
+    Location.{ loc_start = {pos_fname = "test.ml"; pos_lnum = 1; pos_bol = 0; pos_cnum = 5}
+             ; loc_end = {pos_fname = "test.ml"; pos_lnum = 1; pos_bol = 0; pos_cnum = 10}
+             ; loc_ghost = false
+             }
+end
+
+module Factories = struct
+  let ast_name_node ?(loc=Fixtures.location) name =
+    {txt = name; loc }
+
+  let ast_type_node type_name =
+    { ptyp_desc = Ptyp_constr ((ast_name_node (Lident type_name)),  [])
+    ; ptyp_loc = Fixtures.location
+    ; ptyp_attributes = []
+    }
+
+  let ast_constructor_node ~name ~pcd_args ~pcd_res =
+    { pcd_name = ast_name_node name
+    ; pcd_args
+    ; pcd_res
+    ; pcd_loc = Fixtures.location
+    ; pcd_attributes = []
+    }
+end
+
+
+let test_constructor_is_bare =
+  let open Parsetree in
+  let test ~constructor ~expected ctxt =
+    let actual = Ppx_enum_lib.Utils.constructor_is_bare constructor in
+    assert_equal ~ctxt ~cmp:[%eq: bool] ~printer:[%show: bool] expected actual
+  in
+  (* This list of possible formulations is based on the comments in parsetree.mli.
+   * However, I was unable to get the last entry in that list (`| C of {...} as t`)
+   * to compile.*)
+  "constructor_is_bare" >:::
+  [ "| C" >::
+      test
+        ~constructor: (
+          Factories.ast_constructor_node
```

Comment:
You might be able to simplify the test code by taking a `structure_item` as input so that you can use `ppx_metaquot`.
You'd then pass something like:
```ocaml
[%stri type t = A]
```
or
```ocaml
[%stri type t = A of int]
```

Not sure that would actually be an improvement but it is a possibility!

Reply:
To be honest I'm not sure this would be clearer - I think its a bit strand to be passing a type and then converting to a constructor unless there is a nice way to do it in ppxlib

Reply:
Yeah you're right, probably not!